### PR TITLE
Remove deprecated pgk_resources

### LINKED ISSUE #8742: Remove deprecated pgk_resources from everest
### What happened? (You can include a screenshot if it helps explain)

Everest still uses pgk_resources:
```
rg -i pkg_resources
src/everest/templates/__init__.py
1:import pkg_resources
11:    return pkg_resources.resource_filename("everest.templates", template_name)

src/everest/detached/__init__.py
11:import pkg_resources
363:_EVERSERVER_JOB_PATH = pkg_resources.resource_filename(

src/everest/jobs/__init__.py
3:import pkg_resources
32:        return pkg_resources.resource_filename("everest.jobs", rel_script_path)
```

This is deprecated and should therefore be removed, see: https://setuptools.pypa.io/en/latest/pkg_resources.html

### PR DESCRIPTION
**Pull request by frode-aarstad**
**Issue**
Resolves #8742 

**Approach**
_Short description of the approach_

(Screenshot of new behavior in GUI if applicable)

- [ ] PR title captures the intent of the changes, and is fitting for release notes.
- [ ] Added appropriate release note label
- [ ] Commit history is consistent and clean, in line with the [contribution guidelines](https://github.com/equinor/ert/blob/main/CONTRIBUTING.md).
- [ ] Make sure unit tests pass locally after every commit (`git rebase -i main
      --exec 'pytest tests/unit_tests -n logical -m "not integration_test"'`)

## When applicable
- [ ] **When there are user facing changes**: Updated documentation
- [ ] **New behavior or changes to existing untested code**: Ensured that unit tests are added (See [Ground Rules](https://github.com/equinor/ert/blob/main/CONTRIBUTING.md#ground-rules)).
- [ ] **Large PR**: Prepare changes in small commits for more convenient review
- [ ] **Bug fix**: Add regression test for the bug
- [ ] **Bug fix**: Create Backport PR to latest release

### CODE REVIEW

File: src/everest/templates/__init__.py [hunk 1]
```diff
@@ -1,11 +1,17 @@
-import pkg_resources
+import importlib
+from pathlib import Path
 
 template_names = (
+    "schmerge.tmpl",
```
Comment (oyvindeide) [inline src/everest/templates/__init__.py:5]: Why is this needed?

**Review by oyvindeide — APPROVED** [review]
Small question, other than that, LGTM!

File: src/everest/templates/__init__.py (hunk 1 above)
Comment (frode-aarstad) [reply src/everest/templates/__init__.py:5]: I dont know why the template_names list is needed. Either it should include all templates in the dir or we should remove it
